src/main/java/frc/robot/subsystems/swerve/PhoenixOdometryThread.java

```diff
@@ -51,22 +53,40 @@ public static PhoenixOdometryThread getInstance() {
   private PhoenixOdometryThread() {
     setName("PhoenixOdometryThread");
     setDaemon(true);
-    start();
+  }
+
+  @Override
+  public void start() {
+    if (timestampQueues.size() > 0) {
+      super.start();
+    }
   }
 
   public Queue<Double> registerSignal(ParentDevice device, StatusSignal<Double> signal) {
-    isCANFD = CANBus.isNetworkFD(device.getNetwork());
-    Queue<Double> queue = new ArrayBlockingQueue<>(100);
+    Queue<Double> queue = new ArrayDeque<>(100);
     signalsLock.lock();
+    SwerveSubsystem.odometryLock.lock();
     try {
+      isCANFD = CANBus.isNetworkFD(device.getNetwork());
```

Comment:
This happens once for each signal registered but is a global value, so it's going to overwrite previous values. Is that intended? Does this need to be tracked per signal registered?

Reply:
also not sure but it probably shouldn't be overwritten?

Reply:
okay so from my understanding this is used when the thread is run. if it's false (if any of the signals are not CAN FD) it waits and then refreshes all signals and if it's true it just waits for new data, so yes it seems like it should be overwriting. however, if one call sets it as false and then a later call sets it as true before running the thread I don't know if it would handle that properly.